### PR TITLE
TYP: testing.pyi

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
gentle ping @WillAyd, annotations PRs are piling up and becoming blockers (albeit mostly for further annotations, so no real hurry)

File: pandas/_libs/testing.pyi [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+
+def assert_dict_equal(a, b, compare_keys: bool = ...): ...
+
+def assert_almost_equal(a, b,
+                        rtol: float = ..., atol: float = ...,
+                        check_dtype: bool = ...,
```
Comment (WillAyd) [inline pandas/_libs/testing.pyi:7]: I think check_dtype should just be Union[bool, str] right?

Comment (jbrockmendel) [reply pandas/_libs/testing.pyi:7]: in the .pyx file its typed as bint, which i think is better anyway; better to make the casting explicit

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jbrockmendel** [comment]
booyah!   more of these .pyi files are waiting in the wings once the current crop of TYP PRs go through

**Pull request merged**